packages/node/src/methods/install-operations.ts

```diff
@@ -82,8 +104,175 @@ export async function addAppInstance(
+  store: Store,
+  params: Node.ProposeInstallParams
+): Promise<string> {
+  const appInstanceId = generateUUID();
+  const channel = await getChannelFromPeerAddress(
+    selfAddress,
+    params.peerAddress,
+    store
+  );
+
+  const proposedAppInstance = new ProposedAppInstanceInfo(
+    appInstanceId,
+    params
+  );
+
+  await store.addAppInstanceProposal(channel, proposedAppInstance);
+  return appInstanceId;
+}
+
+async function install(
+  store: Store,
+  params: Node.InstallParams
+): Promise<AppInstanceInfo> {
+  if (!params.appInstanceId) {
+    return Promise.reject("No AppInstanceId specified to install");
+  }
+
+  const stateChannel = await store.getChannelFromAppInstanceID(
+    params.appInstanceId
+  );
+
+  const appInstanceInfo = await store.getProposedAppInstanceInfo(
+    params.appInstanceId
+  );
+  const appInstance: AppInstance = createAppInstanceFromAppInstanceInfo(
+    appInstanceInfo,
+    stateChannel
+  );
+  delete appInstanceInfo.initialState;
+
+  const updatedStateChannel = stateChannel.installApp(
+    appInstance,
+    appInstanceInfo.myDeposit,
+    appInstanceInfo.peerDeposit
+  );
+
+  await store.updateChannelWithAppInstanceInstallation(
+    updatedStateChannel,
+    appInstance,
+    appInstanceInfo
+  );
+
+  return appInstanceInfo;
+}
+
+/**
+ * @param appInstanceInfo The AppInstanceInfo to convert
+ * @param channel The channel the AppInstanceInfo belongs to
+ */
+function createAppInstanceFromAppInstanceInfo(
+  proposedAppInstanceInfo: ProposedAppInstanceInfo,
+  channel: StateChannel
+): AppInstance {
+  const appFunctionSigHashes = getAppFunctionSigHashes(proposedAppInstanceInfo);
+
+  const appInterface: AppInterface = {
+    addr: proposedAppInstanceInfo.appId,
+    applyAction: appFunctionSigHashes.applyAction,
+    resolve: appFunctionSigHashes.resolve,
+    getTurnTaker: appFunctionSigHashes.getTurnTaker,
+    isStateTerminal: appFunctionSigHashes.isStateTerminal,
+    stateEncoding: proposedAppInstanceInfo.abiEncodings.stateEncoding,
+    actionEncoding: proposedAppInstanceInfo.abiEncodings.actionEncoding
+  };
+
+  // TODO: throw if asset type is ETH and token is also set
+  const terms: Terms = {
+    assetType: proposedAppInstanceInfo.asset.assetType,
+    limit: proposedAppInstanceInfo.myDeposit.add(
+      proposedAppInstanceInfo.peerDeposit
+    ),
+    token: proposedAppInstanceInfo.asset.token
+      ? proposedAppInstanceInfo.asset.token
+      : AddressZero
+  };
+
+  return new AppInstance(
+    channel.multisigAddress,
+    // TODO: generate ephemeral app-specific keys
+    channel.multisigOwners,
+    proposedAppInstanceInfo.timeout.toNumber(),
+    appInterface,
+    terms,
+    // TODO: pass correct value when virtual app support gets added
+    false,
+    // TODO: this should be thread-safe
+    channel.numInstalledApps,
+    channel.rootNonceValue,
+    proposedAppInstanceInfo.initialState,
+    0,
+    proposedAppInstanceInfo.timeout.toNumber()
```

Comment:
@snario also I'm pretty sure setting this field to be this value isn't right. What should the `latestTimeout` be for an `AppInstance` being instantiated?

Reply:
Yeah it is kind of ambiguous... it is the value for which the most recent signed state update was signed as. If the most recent state update does not exist (this is an install) then the value should probably be the default timeout value.

We're waiting on @IIIIllllIIIIllllIIIIllllIIIIllllIIIIll to define the parameters in the install protocol for defining initial state too, which I think should include state, nonce, and timeout (this val).

Reply:
I'll leave this for now to unblock merge.